labs/6-secdesign/Host/Domain/Services/AccessControlService.cs

```diff
@@ -0,0 +1,92 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Security.Claims;
+using System.Security.Principal;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Caching.Distributed;
+using SecureByDesign.Host.Domain.Model;
+
+namespace SecureByDesign.Host.Domain.Services
+{
+    public class AccessControlService : IAccessControlService
+    {
+        IDistributedCache _permissionsCache;
```

Comment:
Using an simpler `IMemoryCache`, compared to a distributed service.  The thinking is that it is not worth the effort to distribute the cache compared to the simplicity of a single cache.